snuba/migrations/snuba_migrations/outcomes/0003_outcomes_add_category_and_quantity.py

```diff
@@ -0,0 +1,168 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt, String, DateTime
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from sentry_relay import DataCategory
+
+old_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("times_seen", UInt(64)),
+]
+
+new_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("category", UInt(8, Modifiers(nullable=True))),
+    Column("quantity", UInt(64, Modifiers(nullable=True))),
+    Column("times_seen", UInt(64)),
+]
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds quantity and category columns to outcomes. updates materialized view and hourly table to support
+    category as a new dimension and quantity as a new measure.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("quantity", UInt(32, Modifiers(nullable=True))),
+                after=None,
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("category", UInt(8, Modifiers(nullable=True))),
+                after=None,
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_hourly_local",
+                column=Column("quantity", UInt(64, Modifiers(nullable=True))),
+                after=None,
+            ),
+            operations.RunSql(
```

Comment:
not sure if I should have extended `AddColumn` for this case, seems quite specific. Let me know if that is preferable

Reply:
You could add something new like `AddColumnModifyOrderBy` since it's functionally quite different to a plain AddColumn. Up to you though.